### PR TITLE
CR - 2

### LINKED ISSUE #33: CR #2
- [ ] EventManager should not be a singleton
- [ ] Rename NBSSong to NBSSequence or NBSNoteSequence or something
- [ ] Move state out of NBSSong - make the class an immutable record
- [ ] Unused type arg P on AbstractSong - add suppress warn
- [ ] Add KyoriPlatform marker iface
- [ ] Rename SongTick#future to completionSignal

### PR DESCRIPTION
**Pull request by IdanKoblik**
Fix all issues in CR - 2

- [x] EventManager should not be a singleton
- [x] Rename NBSSong to NBSSequence or NBSNoteSequence or something
- [x] Move state out of NBSSong - make the class an immutable record
- [x] Unused type arg P on AbstractSong - add suppress warn
- [x] Add KyoriPlatform marker iface
- [x] Rename SongTick#future to completionSignal

Closes #33

### CODE REVIEW

File: shared/src/main/java/io/github/idankoblik/jukebox/NBSSequence.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        @ApiStatus.AvailableSince("0.0.2") String description,
+        float tempo,
+        List<NBSNote> notes) {
+
+        /**
+         * Gets an unmodifiable list of notes in the song.
+         *
+         * @return a list of {@link NBSNote} objects representing the notes in the song.
+         */
+        public List<NBSNote> getNotes() {
```
Comment (liorsl) [inline shared/src/main/java/io/github/idankoblik/jukebox/NBSSequence.java:29]: There should be a way to do that in construction

File: adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        this.sounds.put((byte) 3, Key.key(NoteInstrument.SNARE.getName()));
+        this.sounds.put((byte) 4, Key.key(NoteInstrument.HAT.getName()));
+        this.sounds.put((byte) 5, Key.key(NoteInstrument.GUITAR.getName()));
+        this.sounds.put((byte) 6, Key.key(NoteInstrument.FLUTE.getName()));
+        this.sounds.put((byte) 7, Key.key(NoteInstrument.BELL.getName()));
+        this.sounds.put((byte) 8, Key.key(NoteInstrument.CHIME.getName()));
+        this.sounds.put((byte) 9, Key.key(NoteInstrument.XYLOPHONE.getName()));
+    }
+
+    @Override
```
Comment (IdanKoblik) [inline adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java:39]: Javadoc

File: adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    public void addInstrument(byte b, @NotNull Key sound) {
+        this.sounds.put(b, sound);
+    }
+
+    @Override
+    public Key removeInstrument(byte b) {
```
Comment (IdanKoblik) [inline adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java:45]: Javadoc

File: adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        return this.sounds.remove(b);
+    }
+
+    @Override
+    public Optional<Key> getInstrument(byte b) {
```
Comment (IdanKoblik) [inline adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java:50]: Javadoc

File: adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        return Optional.ofNullable(this.sounds.get(b));
+    }
+
+    @Override
+    public void clear() {
```
Comment (IdanKoblik) [inline adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java:55]: Javadoc

File: adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        this.sounds.clear();
+    }
+
+    @Override
+    public void loadInstruments(Map<Byte, Key> map) {
```
Comment (IdanKoblik) [inline adventure/src/main/java/io/github/idankoblik/jukebox/manager/KyoriInstrumentManagerImpl.java:60]: Javadoc

File: shared/src/main/java/io/github/idankoblik/jukebox/Playlist.java [hunk 7]
```diff
@@ -11,12 +11,10 @@ public class Playlist<P extends Platform> {
...
     private final Queue<AbstractSong<P>> songs = new LinkedList<>();
     private boolean playing = false;
-
     
     public void addSong(AbstractSong<P> song) {
         songs.offer(song);
     }
-
     
     public void playSongs() {
```
Comment (IdanKoblik) [inline shared/src/main/java/io/github/idankoblik/jukebox/Playlist.java:19]: Javadoc

File: shared/src/main/java/io/github/idankoblik/jukebox/Playlist.java [hunk 8]
```diff
@@ -11,12 +11,10 @@ public class Playlist<P extends Platform> {
 
     private final Queue<AbstractSong<P>> songs = new LinkedList<>();
     private boolean playing = false;
-
     
     public void addSong(AbstractSong<P> song) {
```
Comment (IdanKoblik) [inline shared/src/main/java/io/github/idankoblik/jukebox/Playlist.java:15]: Javadoc

**Comment by IdanKoblik** [comment]
Missing writerside docs

File: shared/src/main/java/io/github/idankoblik/jukebox/manager/InstrumentManager.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
+package io.github.idankoblik.jukebox.manager;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+import java.util.Optional;
+
+public interface InstrumentManager<T> {
```
Comment (IdanKoblik) [inline shared/src/main/java/io/github/idankoblik/jukebox/manager/InstrumentManager.java:8]: Javadoc

**Review by liorsl — APPROVED** [review]

**Pull request merged**